include/Core/DB/TableManager.php

```diff
@@ -211,7 +211,24 @@ public function fetch_log_items( $request, $per_page, $current_page_no ) {
 
 		if ( isset( $request['s'] ) && is_string( $request['s'] ) && $request['s'] !== '' ) {
 			$search_term = trim( esc_sql( $request['s'] ) );
-			$query_cond .= " WHERE ( to_email LIKE '%$search_term%' OR subject LIKE '%$search_term%' ) ";
+
+			if ( Util\is_advanced_search_term( $search_term ) ) {
+				$predicates = Util\get_advanced_search_term_predicates( $search_term );
+
+				foreach ( $predicates as $column => $email ) {
+					switch ( $column ) {
+						case 'to':
+							$query_cond .= ( empty( $query_cond ) ? ' WHERE ' : ' AND ' ) . "to_email LIKE '%$email%'";
+							break;
+						case 'email':
+							$query_cond .= ( empty( $query_cond ) ? ' WHERE ' : ' AND ' ) . "( to_email LIKE '%$email%' OR subject LIKE '%$email%' ) ";
```

Comment:
If the field is `email`, then we need to check `cc`, `bcc`, `to`, `from` and `replyto`.

Also we need to handle the other fields that I have mentioned in the ticket.

Reply:
@sudar the above comments are fixed.